### PR TITLE
Nrf52 add FICR

### PR DESCRIPTION
**Pull request by ppannuto**
### Pull Request Overview

Adds Factory Information Configuration Registers (FICR) structure access.

I intentionally left the RFC `cfg` commit that added a fix for the preview board and the subsequent removal commit as breadcrumbs for anyone stumbling across this in the future.

### Testing Strategy

Reading some registers on a nrf52 as a possible solution to #657, didn't work out, see discussion there, but the register access may still be useful at some point.

### Documentation Updated

- [x] Kernel: ~~The relevant files in `/docs` have been updated or~~ no updates are required.
- [x] ~~Userland: The application README has been added, updated, or no updates are required.~~

### Formatting

- [X] `make formatall` has been run.

### CODE REVIEW

File: chips/nrf52/src/ficr.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    AAAB = 0x41414142,
+    AABA = 0x41414241,
+    AABB = 0x41414242,
+    AAB0 = 0x41414230,
+    AAE0 = 0x41414530,
+    Unspecified = 0xffffffff,
+}
+
+pub struct FICR {
+    registers: *mut FicrRegisters,
```
Comment (niklasad1) [inline chips/nrf52/src/ficr.rs]: not need for this to be `mut` i.e., change to `const`?

File: chips/nrf52/src/ficr.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+impl FICR {
+    const fn new(base_addr: usize) -> FICR {
+        FICR { registers: base_addr as *mut FicrRegisters }
```
Comment (niklasad1) [inline chips/nrf52/src/ficr.rs]: not need for this to be mut i.e., change to const?

File: chips/nrf52/src/ficr.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    pub fn variant(&self) -> Variant {
+        let regs: &mut FicrRegisters = unsafe { mem::transmute(self.registers) };
```
Comment (niklasad1) [inline chips/nrf52/src/ficr.rs]: prefer deferencing over transmuting
```let regs = unsafe { &*self.registers };```

**Comment by ppannuto** [comment]
Good comments -- thanks, fixed.

**Comment by ppannuto** [comment]
To your high-level question -- I had a notion of fixing errata based on variants (to handle the issue from #657), but in that case, preview hardware has an 'unspecified' variant in the register, so it wasn't actually useful.

I don't have an immediate use case for the FICR registers, but wrote this code, so figured I might as well add it. Someday if we had a need to do something specific to handle formally released hardware variants, this is a step towards one possible path.

**Pull request merged**